docs/sql-migration-guide-upgrade.md

```diff
@@ -128,6 +128,8 @@ license: |
 
   - Since Spark 3.0, if `hive.default.fileformat` is not found in `Spark SQL configuration` then it will fallback to hive-site.xml present in the `Hadoop configuration` of `SparkContext`.
 
+  - Since Spark 3.0, for the better extension for fetching shuffle block, we split the protocol into two parts. The protocol change is transparent for the users without external shuffle service, but for the others, we need to upgrade the external shuffle service jar when you see the error message like `java.lang.UnsupportedOperationException: Unexpected message: FetchShuffleBlocks`. If the Spark version is 2.4 or earlier, but external shuffle service is 3.0, we support this scenario by doing the compatible work for OpenBlocks inside `ExternalShuffleBlockHandler`.
```

Comment:
The migration guide is for end users. Let's refine this and only include the information that for end-users, without talking too much about internals.

Reply:
Got it, refined in 6d7f1f3.